docker/build/devdeps.Dockerfile

```diff
@@ -184,14 +154,14 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
         git ninja-build file \
         python3 python3-pip libpython3-dev \
     && python3 -m pip install --no-cache-dir \
-        lit pytest numpy \
-        fastapi uvicorn pydantic requests llvmlite \
-        pyspelling pymdown-extensions \
+        lit==18.1.4 pytest==8.2.0 numpy==1.26.4 requests==2.31.0 \
+        fastapi==0.111.0 uvicorn==0.29.0 pydantic==2.7.1 llvmlite==0.42.0 \
+        pyspelling==2.10 pymdown-extensions==10.8.1 \
```

Comment:
What's the plan for upgrading these over time?

Reply:
Same as any package dependency: manually and periodically. I can remove the change of pinning more versions here, but in the past we had said we wanted to pin them to not have the build randomly break.